File: src/main/java/kr/mjc/rachel/basicsyntax/ConditionalExpressions.java
package kr.mjc.rachel.basicsyntax;

public class ConditionalExpressions {
  public static int maxOf(int a, int b) {
    return a >= b ? a : b;
  }

  public static void main(String[] args) {
    System.out.println("max of 0 and 42 is " + maxOf(0, 42));
  }
}

File: src/main/kotlin/kr/mjc/jacob/basicsyntax/ConditionalExpressions.kt
/*
 * https://kotlinlang.org/docs/basic-syntax.html#conditional-expressions
 */
package kr.mjc.jacob.basicsyntax

fun maxOf(a: Int, b: Int): Int {
  return if (a > b) {
    a
  } else {
    b
  }
}

fun maxOf2(a: Int, b: Int) = if (a > b) a else b

fun main() {
  println("max of 0 and 42 is ${maxOf2(0, 42)}")
}
